pre-work for ng_package rule

alexeagle opened the pull request:
version stamping and `ts_library`->`ng_module`

mary-poppins commented:
You can preview a141f95 at https://pr22176-a141f95.ngbuilds.io/.

mary-poppins commented:
You can preview c07fcb7 at https://pr22176-c07fcb7.ngbuilds.io/.

mary-poppins commented:
You can preview 0119b03 at https://pr22176-0119b03.ngbuilds.io/.

mary-poppins commented:
You can preview fee1ccc at https://pr22176-fee1ccc.ngbuilds.io/.

mary-poppins commented:
You can preview b60df95 at https://pr22176-b60df95.ngbuilds.io/.

mary-poppins commented:
You can preview fd1034f at https://pr22176-fd1034f.ngbuilds.io/.

mary-poppins commented:
You can preview 345f6de at https://pr22176-345f6de.ngbuilds.io/.

IgorMinar commented:
hmm.. ci is not happy. I think you should cherry-picked/squash https://github.com/angular/angular/pull/21431/commits/c724e7d394c393a248b623d23a7a95af78d16b3b into this PR otherwise people running bazel build will have a bad time.

you will also need to pull tags on CI as instructed in the error message in that PR.

mary-poppins commented:
You can preview f76b7f8 at https://pr22176-f76b7f8.ngbuilds.io/.

alexeagle committed:
refactor(bazel): convert most ts_library to ng_module

This is necessary so we can produce ng metadata for our packages that are published as libraries

alexeagle committed:
feat(bazel): introduce a binary stamping feature

This grabs version control metadata and makes it available in the build, eg. to put in the version field for released artifacts

mary-poppins commented:
You can preview c7f3d11 at https://pr22176-c7f3d11.ngbuilds.io/.

mary-poppins commented:
You can preview 2602b6d at https://pr22176-2602b6d.ngbuilds.io/.

IgorMinar commented:
@alexeagle should we consider upstreaming the version stamping some place outside of angular/angular (or into @angular/bazel) - I expect that @jelbourn and @hansl will also want to use it.

this would be done in a separate PR so that we don't block this one any more.

alexeagle committed:
feat(bazel): ng_module produces bundle index

It creates the bundle index .d.ts and .metadata.json files.
The names are based on the ng_module target.

mary-poppins commented:
You can preview a0bb894 at https://pr22176-a0bb894.ngbuilds.io/.

alexeagle commented:
found some issues using this from `ng_package`, will fix up in this PR, not ready to merge yet...

mary-poppins commented:
You can preview 31a6443 at https://pr22176-31a6443.ngbuilds.io/.

mary-poppins commented:
You can preview d5f3a09 at https://pr22176-d5f3a09.ngbuilds.io/.
